CMakeLists.txt

```
@@ -242,6 +242,7 @@ option(USE_ASAN "Use Address+Undefined Sanitizers" OFF)
 option(USE_TSAN "Use Thread Sanitizer" OFF)
 option(USE_CUDA "Use CUDA" ON)
 option(USE_XPU "Use XPU" ON)
+option(NO_DEPRECATED "Don't compile deprecated code" OFF)
```

Nit (as it this is just about macro definitions, isn't it?)
```suggestion
option(NO_DEPRECATED "Don't defined deprecated macro" OFF)
```